modules/trusted-firmware-m/tfm_boards/include/zephyr/kernel.h

```diff
@@ -7,16 +7,103 @@
 #ifndef __ZEPHYR_KERNEL_H
 #define __ZEPHYR_KERNEL_H
 
-/* Compatebility header for using Zephyr API in TF-M.
+/*
+ * Compatibility header for using Zephyr API in TF-M.
  *
  * The macros and functions here can be used by code that is common for both
  * Zephyr and TF-M RTOS.
  *
  * The functionality will be forwarded to TF-M equivalent of the Zephyr API.
+ *
  */
 
+#include <stdint.h>
+#include <stdbool.h>
 #include <zephyr/sys/printk.h>
 
+#include <zephyr/kernel_includes.h>
+#include <errno.h>
+#include <stdbool.h>
+
 #define k_panic() tfm_core_panic()
 
+/*
+ * This function is only used by cryptomaster to determine if it can
+ * use concurrency primitives. In the TF-M build we want to always use
+ * synchronization primitives so we return 0.
+ */
+static inline bool k_is_pre_kernel(void)
+{
+	return 0;
+}
+
+#define K_FOREVER 0
+
+/*
+ * K_MUTEX is only used to enforce single-user access in a
+ * multi-threaded environment and TF-M is a single-threaded
```

Comment:
```suggestion
 * multi-threaded environment and TF-M is currently a single-threaded
```

Reply:
everything is currently. I don't feel this change helps understand the code.